app/repositories/s3_enrolment_repo.py

```diff
@@ -0,0 +1,107 @@
+import json
+from typing import Any, List
+
+import boto3
+
+from app.config import settings
+from app.domain.entities.enrolment_request import (
+    VALID_STATE_CHANGES,
+    EnrolmentFilters,
+    EnrolmentRequest,
+)
+from app.repositories.enrolment_repo import EnrolmentRepo
+from app.utils.error_handling import handle_s3_errors
+
+
+class S3EnrolmentRepo(EnrolmentRepo):
+    s3: Any
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.s3 = boto3.client("s3", **settings.s3_configuration)
+
+    def get_enrolments(
+        self, enrolment_filters: EnrolmentFilters
+    ) -> List[EnrolmentRequest]:
+        enrolment_filters = enrolment_filters.dict()
+        with handle_s3_errors():
+            enrolment_objects_list = self.s3.list_objects(
+                Bucket=settings.ENROLMENT_BUCKET
```

Comment:
you are getting requests from enrl bucket? that is used by emp-admin
I named it ENROLMENT_AUTHORISATION_BUCKET in TP-enrl
lets decide a name for it.

Reply:
I think your naming is good. I'll make changes for that.